docs/configuration/configuration.md

```diff
@@ -6,17 +6,18 @@ highest priority, followed by environment variables and finally config file.
 Supported flags are as follows (result of running `./flagd start --help`):
 
 ```
-  -b, --bearer-token string       Set a bearer token to use for remote sync
-  -e, --evaluator string          Set an evaluator e.g. json (default "json")
-  -h, --help                      help for start
-  -p, --port int32                Port to listen on (default 8013)
-  -m, --metrics-port int32        Port to set up metrics listener on (default 8014)
-  -c, --server-cert-path string   Server side tls certificate path
-  -k, --server-key-path string    Server side tls key path
-  -a, --sync-provider-args        Sync provider arguments as key values separated by =
-  -d, --socket-path string        Set the flagd socket path.
-  -f, --uri strings               Set a sync provider uri to read data from this can be a filepath, url or reference to a kubernetes custom resource. Using multiple providers is supported, however if flag keys are duplicated across multiple sources it may lead to unexpected behavior.
-  -C, --cors-origin strings       Set a CORS allow origin header, setting "*" will allow all origins (by default CORS headers are not set)
+  -b, --bearer-token string                 Set a bearer token to use for remote sync
+  -C, --cors-origin strings                 CORS allowed origins, * will allow all origins
+  -e, --evaluator string                    Set an evaluator e.g. json, yaml/yml (default "json")
+  -h, --help                                help for start
+  -m, --metrics-port int32                  Port to serve metrics on (default 8014)
+  -p, --port int32                          Port to listen on (default 8013)
+  -c, --server-cert-path string             Server side tls certificate path
+  -k, --server-key-path string              Server side tls key path
+  -d, --socket-path string                  Flagd socket path. With grpc the service will become available on this address. With http(s) the grpc-gateway proxy will use this address internally.
+  -y, --sync-provider string                DEPRECATED: Set a sync provider e.g. filepath or remote
```

Comment:
It looks like the state of these configuration options have drifted from `main`. For example, `-y` is no longer an option.

Reply:
@vadasambar I think the only change needed to this file is to mention the yaml/json evaluations work the same, and yaml/json format are auto-detected by the file extension.

Reply:
@skyerus it is still an option (although `DEPRECATED`)

https://github.com/open-feature/flagd/blob/af6f133688b506f52ac6180890e3d785238d7b40/cmd/start.go#L54

If you are suggesting we remove a deprecated option from the docs, I am not sure if that's such a good idea. If you think I am misunderstanding what you are seeing, please feel free to correct me.

Reply:
@toddbaert thank you for the feedback. I have taken some liberty in re-wording your suggestion like this:
```
I think the only change needed to this file is to mention the yaml/json evaluations work the same
```
to
```
Please note that yaml/yml and json evaluations work the same (yaml/yml files are converted to json internally)
```
I added extra context. 

As for
```
yaml/json format are auto-detected by the file extension.
```
I have re-worded it to (and put it next it `-f` flag because I think the end-user would look at `-f` flag help for such information)
```
Please note that if you are using filepath, flagd only supports files with .yaml/.yml/.json extension.
```
This is because I thought `auto-detected by file extension` seemed a little vague to the end-user who probably won't be as well-versed in the internals of flagd. The problem with my sentence is, it doesn't explain the "why" but your suggestion does. 

This is not final. If you have any opinions, please let me know :pray: I am open to changing things.

Reply:
Yes, seems @vadasambar  needs a rebase after a fresh upstream sync. As you can see in the current master version [1], we no longer have -y option. And no, @skyerus  is not saying you should remove that, but rather pointing to you that you have diverted from upstream (flagd main).

If you go to your fork [https://github.com/vadasambar/flagd/tree/main] , you can see it says 

> This branch is [13 commits behind](https://github.com/vadasambar/flagd/compare/main...open-feature:flagd:main) open-feature:main.

which is the root cause.

@vadasambar, can you perform the following, 

1. Update your fork's [https://github.com/vadasambar/flagd] main branch from the latest flagd main [https://github.com/open-feature/flagd]. 

This is easy and doable through Git UI. Check the following screenshot for "Sync fork" feature

![image](https://user-images.githubusercontent.com/8186721/209706358-18a2b173-e330-41ee-8aa4-905b360817d2.png)


2. Once the above step is done, rebase this branch on your fork's main OR Merge your fork's master to this branch if you are  familiar using merge instead of rebases 

This should resolve the conflict we see here.

[1] https://github.com/open-feature/flagd/blob/main/docs/configuration/configuration.md?plain=1

Reply:
@Kavindu-Dodan I haven't updated my `origin` but I don't think it's related (I pull the latest from `upstream` `main` in my branch which is why you don't see weird diffs in this PR).  

I checked `main` of https://github.com/open-feature/flagd before posting my comment. I can still see the `-y` option in the code in the `main` branch. 

Here's a screenshot:
![image](https://user-images.githubusercontent.com/34534103/209765416-b82bb643-01bf-4015-ba74-9a2ffa1a8487.png)

Reply:
Yes and that option is in a "deprecation" state. You can check this PR which contains the change - https://github.com/open-feature/flagd/pull/222 . And PR removed the configuration entry. 

Further, the most recent change to this file is done through this commit -  https://github.com/open-feature/flagd/commit/d6147490fd939f227025b2875b2a1888b4e4a3fe  . This is why you need to sync your fork and then update your PR so it reflect recent changes.

Reply:
I don't think we should [remove a flag from docs](https://github.com/open-feature/flagd/pull/222/files#diff-17ed18489a956f326ec0fe4040850c5bc9261d4631fb42da4c52891d74a59180L18) *just because it is deprecated* (not a good user experience). That is why I have added it back. I haven't seen a tool/software I have used or worked with removing docs for deprecated flags/features until the flag was completely removed (including Kubernetes). 

> While a deprecated software feature remains in the software, its use may raise warning messages recommending alternative practices. Deprecated status may also indicate the feature will be removed in the future. Features are deprecated, rather than immediately removed, to provide backward compatibility and to give programmers time to bring affected code into compliance with the new standard. 

https://en.wikipedia.org/wiki/Deprecation
TL:DR; deprecation is not equal to removal. 

> Further, the most recent change to this file is done through this commit - https://github.com/open-feature/flagd/commit/d6147490fd939f227025b2875b2a1888b4e4a3fe . This is why you need to sync your fork and then update your PR so it reflect recent changes.

You are looking in the wrong place? Check https://github.com/vadasambar/flagd/commits/feat/180/add-yaml-evaluator. Maybe I am not explaining it well. Here's what I did:
```
# I am on my git branch vadasambar:feat/180/add-yaml-evaluator
$ git pull upstream main --rebase
....
```
My upstream points to github.com/open-feature/flagd
```
$ git remote -v
origin  git@github.com:vadasambar/flagd.git (fetch)
origin  git@github.com:vadasambar/flagd.git (push)
upstream        git@github.com:open-feature/flagd.git (fetch)
upstream        git@github.com:open-feature/flagd.git (push)
```
Because of which my `main` branch in my fork i.e., https://github.com/vadasambar/flagd is behind https://github.com/open-feature/flagd repo's `main` branch which is *okay* but my `vadasambar:feat/180/add-yaml-evaluator` is *up-to-date*. 

I have done something analogous to https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork#syncing-a-fork-branch-from-the-command-line but my local branch is `vadasambar:feat/180/add-yaml-evaluator`

Reply:
If the flagd community as a whole thinks we should not be keeping docs for `deprecated` flags, I will go ahead and remove it (but I strongly disagree with the practice). I will wait for comments from other reviewers before I remove it.

Reply:
@vadasambar This shows why a PR must focus on its intended purpose - for example, this PR should focus on "adds support for YAML evaluator" as in the description of the PR. My assumption was that your branch diverted and hence the configuration entry on "-y" option.

IMO, if you think this is a bad practice and the flag should still exist, then it should have been added through a dedicated PR with a dedicated description/justification. This way, we can maintain a clear Git history, have focused reviews & discussions :)

Reply:
I think I agree with @vadasambar that it actually makes sense to keep this doc and mention it's deprecated, though it's not really the original focus of this PR.

In the future I would generally recommend to open new issues about such things to make things easier for everyone and keep discussions separate. I personally really want this yaml feature added and don't want it to be blocked any longer than necessary for the somewhat unreleated `--sync provider` arg deprecation discussion.

However, since the work has already been done on this branch I'm fine with it.